src/discovery/Discovery.ts

```diff
@@ -11,252 +12,400 @@ import type {
-            const linkedVertexNodeInfo: NodeInfo = {
-              id: nodesUtils.encodeNodeId(linkedVertexNodeId),
-              chain: linkedVertexChainData,
+      if (!(await this.queueIsEmpty())) {
+        for await (const o of this.discoveryQueueDb.createReadStream()) {
+          const vertexId = IdInternal.fromBuffer(o.key) as DiscoveryQueueId;
+          const data = o.value as Buffer;
+          const vertex = await this.db.deserializeDecrypt<GestaltKey>(
+            data,
+            false,
+          );
+          const vertexGId = gestaltsUtils.ungestaltKey(vertex);
+          if (vertexGId.type === 'node') {
+            // The sigchain data of the vertex (containing all cryptolinks)
+            let vertexChainData: ChainData = {};
+            // If the vertex we've found is our own node, we simply get our own chain
+            const nodeId = nodesUtils.decodeNodeId(vertexGId.nodeId)!;
+            if (nodeId.equals(this.nodeManager.getNodeId())) {
+              const vertexChainDataEncoded =
+                await this.nodeManager.getChainData();
+              // Decode all our claims - no need to verify (on our own sigchain)
+              for (const c in vertexChainDataEncoded) {
+                const claimId = c as ClaimIdEncoded;
+                vertexChainData[claimId] = claimsUtils.decodeClaim(
+                  vertexChainDataEncoded[claimId],
+                );
+              }
+              // Otherwise, request the verified chain data from the node
+            } else {
+              vertexChainData = await this.nodeManager.requestChainData(nodeId);
+            }
+            // TODO: for now, the chain data is treated as a 'disjoint' set of
+            // cryptolink claims from a node to another node/identity
+            // That is, we have no notion of revocations, or multiple claims to the
+            // same node/identity. Thus, we simply iterate over this chain of
+            // cryptolinks.
+            // Now have the NodeInfo of this vertex
+            const vertexNodeInfo: NodeInfo = {
+              id: nodesUtils.encodeNodeId(nodeId),
+              chain: vertexChainData,
             };
-            await this.gestaltGraph.linkNodeAndNode(
-              vertexNodeInfo,
-              linkedVertexNodeInfo,
-            );
-
-            // Add this vertex to the queue if it hasn't already been visited
-            const linkedVertexGK =
-              gestaltsUtils.keyFromNode(linkedVertexNodeId);
-            if (!visitedVertices.has(linkedVertexGK)) {
-              vertexQueue.push(linkedVertexGK);
+            // Iterate over each of the claims in the chain (already verified)
+            // TODO: because we're iterating over keys in a record, I don't believe
+            // that this will iterate in lexicographical order of keys. For now,
+            // this doesn't matter though (because of the previous comment).
+            for (const claimId in vertexChainData) {
```

Comment:
This is directly related to #327. We don't currently have the need for ordering in our claims because our claims are only additions. But obviously this is going to be extended - so will be necessary for #327 to be completed before we can support revocations with discovery. I imagine that revocations will require this discovery process to iterate over claims in reverse chronological order.

Nothing to resolve in this PR though, just something to point out for future.

Reply:
Ok, I've resolve this then.